runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -3735,6 +3736,15 @@ struct amRequest_AMO_t {
   void* result;                 // result address on initiator's node
 };
 
+struct amRequest_FAMO_result_t {
+  struct amRequest_base_t b;
+  int8_t size;                  // object size (bytes)
+  chpl_amo_datum_t object;      // the object
+  enum fi_datatype ofiType;     // ofi object type
+  void* obj;                    // object address on target node
```

Comment:
Also, I think I might call this `res` instead of `obj`. I think of `obj` as the object (variable) of the atomic op. What we're sending back here is the result value, i.e. the pre-AMO value that was in the object variable.